libs/azure/odc/azure/__init__.py

```diff
@@ -0,0 +1,56 @@
+"""Azure blob storage crawling and YAML fetching utilities
+"""
+from azure.storage.blob import ContainerClient, BlobClient
+from typing import List, Tuple, Optional
+from multiprocessing.dummy import Pool as ThreadPool
+from functools import partial
+
+def find_blobs(account_url: str, container_name: str, credential: str, prefix: str, suffix: str):
```

Comment:
I would make this into iterator, so `yield blob_name` instead of `blob_list.append(..)`.
You can still make list out of it with `list(find_blobs(...))` if that's what's needed later on.

Also missing return type ` -> List[str]` with the current code, or `-> Iterator[str]` if changing to iterator.